generators/app/index.js

```diff
@@ -1,49 +1,91 @@
-const fs = require('fs')
-const colors = require('colors')
-const Generator = require('yeoman-generator')
-const upath = require('upath')
-const appSettings = require('../../package.json')
+const fs = require('fs');
+const Generator = require('yeoman-generator');
+const upath = require('upath');
+const header = require('./header');
 
 module.exports = class extends Generator {
+
   constructor (args, opts) {
-    super(args, opts)
-    const year = new Date().getYear() + 1900
-    this.log('/------------------------------------------/'.cyan)
-    this.log('/'.cyan + '        SYMPHONY GENERATOR  '.bold + appSettings.version.bold + '         /'.cyan)
-    this.log('/    by platformsolutions@symphony.com     /'.cyan)
-    this.log(`/ (c) ${year} Symphony Communication Services /`.cyan)
-    this.log('/------------------------------------------/'.cyan)
+    super(args, opts);
   }
 
-  init () {
-
+  /**
+   * Displays the Welcome header (current version, year, etc.)
+   */
+  initializing () {
+    header.display(this.log);
   }
 
+  /**
+   * Displays the initial menu (application type, language, auth, etc.)
+   */
   prompting () {
-    return this.prompt([
+
+    const modules = {
+      application: {
+        node: {
+          path: '../node-ext-apps'
+        },
+        java: {
+          path: '../java-ext-apps'
+        }
+      },
+      bot: {
+        node: {
+          path: '../node-bots'
+        },
+        java: {
+          path: '../java'
+        },
+        dotnet: {
+          path: '../dotnet-bots'
+        },
+        python: {
+          path: '../python-bots'
+        }
+      }
+    };
+
+    const projectType = {
```

Comment:
In general I think that I would rather define these variables in a separate file or at least outside of the class (like above the class).
Otherwise it's not easy to find these constants when looking for them, if they are only defined in the implementation of the function.

Reply:
Done